test/plugin/test_out_google_cloud.rb

```diff
@@ -1295,140 +180,59 @@ def setup_logging_stubs
-                            'Content-Type' => 'application/json' })
-  end
-
-  def setup_managed_vm_metadata_stubs
-    stub_metadata_request(
-      'instance/attributes/',
-      "attribute1\ngae_backend_name\ngae_backend_version\nlast_attribute")
-    stub_metadata_request('instance/attributes/gae_backend_name',
-                          MANAGED_VM_BACKEND_NAME)
-    stub_metadata_request('instance/attributes/gae_backend_version',
-                          MANAGED_VM_BACKEND_VERSION)
-  end
-
-  def setup_container_metadata_stubs
-    stub_metadata_request(
-      'instance/attributes/',
-      "attribute1\nkube-env\nlast_attribute")
-    stub_metadata_request('instance/attributes/kube-env',
-                          "ENABLE_NODE_LOGGING: \"true\"\n"\
-                          'INSTANCE_PREFIX: '\
-                          "gke-#{CONTAINER_CLUSTER_NAME}-740fdafa\n"\
-                          'KUBE_BEARER_TOKEN: AoQiMuwkNP2BMT0S')
+  def create_driver(conf = APPLICATION_DEFAULT_CONFIG, tag = 'test')
+    Fluent::Test::BufferedOutputTestDriver.new(
+      Fluent::GoogleCloudOutput, tag).configure(conf, true)
   end
 
-  def setup_cloudfunctions_metadata_stubs
-    stub_metadata_request(
-      'instance/attributes/',
-      "attribute1\nkube-env\ngcf_region\nlast_attribute")
-    stub_metadata_request('instance/attributes/kube-env',
-                          "ENABLE_NODE_LOGGING: \"true\"\n"\
-                          'INSTANCE_PREFIX: '\
-                          "gke-#{CLOUDFUNCTIONS_CLUSTER_NAME}-740fdafa\n"\
-                          'KUBE_BEARER_TOKEN: AoQiMuwkNP2BMT0S')
-    stub_metadata_request('instance/attributes/gcf_region',
-                          CLOUDFUNCTIONS_REGION)
+  # The caller can optionally provide a block which is called for each entry.
+  def verify_log_entries(n, params, payload_type = 'textPayload', &block)
+    verify_json_log_entries(n, params, payload_type, &block)
   end
 
-  def container_log_entry_with_metadata(log)
-    {
-      log: log,
-      stream: CONTAINER_STREAM,
-      time: CONTAINER_TIMESTAMP,
-      kubernetes: {
-        namespace_id: CONTAINER_NAMESPACE_ID,
-        namespace_name: CONTAINER_NAMESPACE_NAME,
-        pod_id: CONTAINER_POD_ID,
-        pod_name: CONTAINER_POD_NAME,
-        container_name: CONTAINER_CONTAINER_NAME,
-        labels: {
-          CONTAINER_LABEL_KEY => CONTAINER_LABEL_VALUE
-        }
-      }
-    }
+  # For an optional field with default values, Protobuf omits the field when
+  # deserialize it to json. So we need to add an extra check for gRPC which uses
+  # Protobuf.
+  #
+  # An optional block can be passed in if we need to assert something other than
+  # a plain equal. e.g. assert_in_delta.
+  def assert_with_default_check(field, expected_value, _default_value, entry)
+    if expected_value == 'DEBUG'
+      # For some reason we return '100' instead of 'DEBUG' for the non-grpc
```

Comment:
This seems to be a bug in the test, introduced when moving code from this file to `base_test.rb`. Let's get rid of this.